firebase-firestore/src/main/java/com/google/firebase/firestore/remote/RemoteSerializer.java

```diff
@@ -643,9 +643,7 @@ private List<Filter> decodeFilters(StructuredQuery.Filter proto) {
     Filter result = decodeFilter(proto);
 
     // Instead of a singletonList containing AND(F1, F2, ...), we can return a list containing F1,
-    // F2, ...
-    // TODO(orquery): Once proper support for composite filters has been completed, we can remove
-    // this flattening from here.
+    // F2, ... to stay consistent with the older SDK versions.
```

Comment:
This TODO was released because we'd like the user to see the same canonical ID for a target even after they upgrade their SDK.